src/java/org/apache/cassandra/index/sai/IndexContext.java

```diff
@@ -448,6 +399,99 @@ public int hashCode()
     public String logMessage(String message)
     {
         // Index names are unique only within a keyspace.
-        return String.format("[%s.%s.%s] %s", keyspace, table, config == null ? "?" : config.name, message);
+        return String.format("[%s.%s.%s] %s", keyspace, table, indexMetadata == null ? "?" : indexMetadata.name, message);
```

Comment:
REMINDER: Let's look at how often this might happen on a hot path and create unnecessary strings.

Reply:
There are 2 `logMessage` implementations, this one and the one in `IndexDescriptor`. I have done an review of the usage of both these methods and have the following observations:

- I couldn't find any use of the methods being used on hot-paths that aren't error conditions or used in protected trace statements.
- The main use of the methods by debug statements are in initialisation / build / write operations and only at start and end of builds / writes. The only use in build row adds is for errors.

I am assuming in the above that the hot path means the happy code paths of reads and writes. I couldn't see non-wrapped usage on these paths.